Modules/_testcapi/complex.c

```diff
@@ -85,6 +88,53 @@ complex_asccomplex(PyObject *Py_UNUSED(module), PyObject *obj)
     return PyComplex_FromCComplex(complex);
 }
 
+static PyObject*
+_py_c_neg(PyObject *Py_UNUSED(module), PyObject *num)
+{
+    Py_complex complex;
+
+    complex = PyComplex_AsCComplex(num);
+    if (complex.real == -1. && PyErr_Occurred()) {
+        return NULL;
+    }
+
+    return PyComplex_FromCComplex(_Py_c_neg(complex));
+}
+
+#define _PY_C_FUNC2(suffix)                                      \
+    static PyObject *                                            \
+    _py_c_##suffix(PyObject *Py_UNUSED(module), PyObject *args)  \
+    {                                                            \
+        Py_complex num, exp, res;                                \
+                                                                 \
+        if (!PyArg_ParseTuple(args, "DD", &num, &exp)) {         \
+            return NULL;                                         \
+        }                                                        \
+                                                                 \
+        errno = 0;                                               \
+        res = _Py_c_##suffix(num, exp);                          \
+        _Py_ADJUST_ERANGE2(res.real, res.imag);                  \
+                                                                 \
+        if (errno == EDOM) {                                     \
+            PyErr_SetString(PyExc_ZeroDivisionError,             \
+                            "complex division by zero");         \
+            return NULL;                                         \
+        }                                                        \
+        else if (errno == ERANGE) {                              \
+            PyErr_SetString(PyExc_OverflowError,                 \
+                            "complex exponentiation");           \
+            return NULL;                                         \
+        }                                                        \
```

Comment:
The wrapper should simply return `res` and `errno`.

Reply:
Hmm, this was for some consistency with ``_py_c_neg()``.  Will redo.